### ISSUE TITLE
Mapping of HealthProfessional::HealthProfessionalRole

### ISSUE THREAD

**Issue by pieter-edelman-nictiz** [Nictiz/Nictiz-R4-zib2020#44]
Zib HealthProfessional defines a concept called HealthProfessionalRole, defined as "The role the health professional fulfils in the healthcare process. For health professionals, this could be for example attender, referrer or performer". In FHIR this is usually specified not in the referred but in the referring profile, to allow re-use of the same Practitioner/PractiotionerRole model. For example, Procedure defines `Procedure.performer.role` for this purpose, and CareTeam specifies `CareTeam.participant.role`. It makes sense to map HealthProfessionalRole on these FHIR elements.
However, not every referring resource defines a role, e.g. Observation just has `Observation.performer` and cannot specify the role of the performer. This means that HealthProfessionalRole cannot always be captured.
Possible solutions:
1. Add a (custom) extension to the PractitionerRole profile to capture HealthProfessionalRole. This would negate the benefit of having a single Health Professional model that can be re-used to some extent.
2. Add a (custom) extension to the reference datatype and add it everywhere where a HealthProfessional is referred. This would involve quite a bit of extra work.

**Comment by ArdonToonstra** [comment]
At first sight, I am not happy with either option at all. You would think there may be some logic in FHIR on the reasons why `.role` elements are skipped in certain resources... 

It might help if we make an analysis on how often a HealthProfessionalRole cannot be captured because there is not a suitable element for it in the FHIR resource.

**Comment by pieter-edelman-nictiz** [comment]
True, but there's also some logic as to why the zib does capture this role. But agreed, let's find out where FHIR omits this role.

**Comment by ArdonToonstra** [comment]
Here are my results based on some preliminary mappings to resources. For the majority of references to HealthProfessional there is no suitable element for HealthProfessionalRole concept. However, the amount of HealthProfessional references made by the zibs is lower than I expected. Therefore, I think I am inclined to go for option 2. Create a custom extension on the reference datatype. We could make a profile on the reference datatype to simplify things a bit?

| Zib                               | FHIR resource            | Mapping voor HealthProfessionalRole                     | Notes                                                                                                                                                                                                                                                                                               |
|-----------------------------------|--------------------------|---------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| CareAgreement-v1.0                | CarePlan                 | None                                                    | Has 3 references to HealthProfessional: CareCoordinator, Performer and AgreementParties                                                                                                                                                                                                             |
| HelpFromOthers-v3.01              | CarePlan                 | None                                                    |                                                                                                                                                                                                                                                                                                     |
| LaboratoryTestResult-v4.6         | Observation              | None                                                    |                                                                                                                                                                                                                                                                                                     |
| MedicalDevice-v3.3.1              | DeviceUseStatement       | None                                                    | The STU3 profile uses a custom extension for the HealthProfessional reference and there is no suitable element in R4 too,-- make it a complex   extension? No definitioncode is given for the context of the reference   therefore I am wondering if the HealthprofessionalRole is applicable here. |
| MedicationAgreement-v1.2          | MedicationRequest        | None                                                    | A .performerType exists, but this is not about the prescriber/requester                                                                                                                                                                                                                             |
| MedicationContraIndication-v1.0   | Flag                     | None                                                    |                                                                                                                                                                                                                                                                                                     |
| MedicationUse2-v1.1.1             | MedicationStatement      | None                                                    | Prescriber is already captured with a custom extension -- we can make it   a complex custom extensino…                                                                                                                                                                                              |
| SOAPReport-v1.0                   | Composition(?)           | None                                                    | .author                                                                                                                                                                                                                                                                                             |
| TreatmentDirective2-v1.0          | Consent                  | None                                                    | This zib changes significantly between 2017 / 2020. The reference was   first mapped to a custom extension but now .performer should be suitable. I   wonder, however, if the HealthprofessionalRole really matters here.                                                                           |
| LaboratoryTestResult-v4.6         | DiagnosticReport         | .performer.role &      ServiceRequest.performerType     |                                                                                                                                                                                                                                                                                                     |
| NursingIntervention-v3.2          | Procedure                | .performer.function &      ServiceRequest.performerType |                                                                                                                                                                                                                                                                                                     |
| MedicationAdministration2-v1.1.1  | MedicationAdministration | .performer.function                                     |                                                                                                                                                                                                                                                                                                     |
| Procedure-v5.2                    | Procedure                | .performer.function                                     |                                                                                                                                                                                                                                                                                                     |
| Vaccination-v4.0                  | Immunization             | .performer.function                                     |                                                                                                                                                                                                                                                                                                     |
| Encounter-v4.0.1                  | Encounter                | .participant.type                                       |                                                                                                                                                                                                                                                                                                     |
| MultidisciplinaryTeamMeeting-v1.0 | Encounter(?)             | .participant.type                                       |                                                                                                                                                                                                                                                                                                     |
| CareTeam-v1.0                     | CareTeam                 | .participant.role                                       |                                                                                                                                                                                                                                                                                                     |

**Comment by pieter-edelman-nictiz** [comment]
A few notes:
* The CarePlan based resources _may_ end up using a reference to a CareTeam instead of inline references to participants, which natively support a role.
* CarePlan might be better suitable than Consent for TreatmentDirective. Therefore, the above remark applies again.
* DiagnosticReport does not have a `.performer.role` in FHIR R4.
* `Procedure.performer.function` seems to specify the function rather than the role of the performer (e.g. "a neurosurgeon" rather than "the surgeon performing the procedure"). For MedicationAdministration and Immunization, `.performer.function` seem to match the zib concept better.

**Comment by ArdonToonstra** [comment]
Thanks, I indeed missed some of those notes... I tried to incorporate them in the above table overview.

**Comment by ArdonToonstra** [comment]
Update: seems like most big hospitals register this concept for Procedures, Encounters and EpisodeOfCare. Therefore I wonder if this concept is actually applicable for all HCIMs in the list above. For example, is the HealthProfessionalRole applicable for the MedicationContraIndication author...

**Comment by pieter-edelman-nictiz** [comment]
Pull request #55 zou dit issue moeten oplossen.

**Comment by ahenket** [comment]
FHIR Core allows for further definition of the type of participation/performer/author, but not everywhere. This seems to align with normal practice. Not every participation needs HealthProfessionalRole /ZorgverlenerRol.

So agree with Ardon to only use the `.role`/`.function` when available in the resource. Note that in http://hl7.org/fhir/R4/valueset-participant-role.html connected to CareTeam.participant.role for example, a case manager is actually represented albeit using the some extension of SNOMED CT.
